setuptools/command/sdist.py

```diff
@@ -126,14 +127,18 @@ def _add_defaults_python(self):
         if self.distribution.has_pure_modules():
             build_py = self.get_finalized_command('build_py')
             self.filelist.extend(build_py.get_source_files())
-            # This functionality is incompatible with include_package_data, and
-            # will in fact create an infinite recursion if include_package_data
-            # is True.  Use of include_package_data will imply that
-            # distutils-style automatic handling of package_data is disabled
-            if not self.distribution.include_package_data:
-                for _, src_dir, _, filenames in build_py.data_files:
-                    self.filelist.extend([os.path.join(src_dir, filename)
-                                          for filename in filenames])
+
+            if self.distribution.include_package_data:
+                # This functionality is incompatible with include_package_data,
```

Comment:
This comment, referencing "this functionality" is no longer proximate to the functionality that it references. Probably you should extract a function that encapsulates the operation so that the scope of the comment is clear.